src/popupmenu.c

```diff
@@ -346,15 +326,8 @@ pum_display(
 			    pum_width = Columns - pum_col - 1;
 		    }
 		}
-		else if (pum_width > max_width + pum_kind_width
-							  + pum_extra_width + 1
-			    && pum_width > p_pw)
-		{
-		    pum_width = max_width + pum_kind_width
-							 + pum_extra_width + 1;
-		    if (pum_width < p_pw)
-			pum_width = p_pw;
-		}
+		else if (pum_width > content_width && pum_width > p_pw)
+		    pum_width = MIN(content_width, p_pw);
```

Comment:
This seems wrong. It should be `MAX()` here.

Reply:
oh..yes

Reply:
fixed in https://github.com/vim/vim/pull/16210

Reply:
Gna, I stumbled over it but thought my reasoning was wrong. I just included just this fix, @glepnir can you please backout this change from #16210 ?

Reply:
sure